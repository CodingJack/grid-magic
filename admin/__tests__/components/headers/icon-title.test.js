/**
 * External dependencies.
 */
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

/**
 * Internal dependencies.
 */
import IconTitle from '../../../src/js/components/headers/icon-title';

const namespace = 'esg-v4';
Enzyme.configure( { adapter: new Adapter() } );

test( 'it renders correctly', () => {
  const tree = create(
    <IconTitle 
      namespace={ namespace } 
      text="Some Text"
      icon="close" 
    />
  ).toJSON();
  
  if( ! tree ) {
    return;
  }
  expect( tree ).toMatchSnapshot();
  const { props } = tree;
  
  props.namespace = 'name-space';
  props.text = '';
  props.icon = 'check'; 
  props.color = 'green';
  props.bold = true;
  props.uppercase = true;

  expect( tree ).toMatchSnapshot();
} );

describe('it unmounts', () => {
  let clicked;
  const component = shallow(
    <IconTitle 
      namespace={ namespace } 
      text="Some Text"
      icon="check" 
      onClick={ () => clicked = true } 
    /> 
  );

  it('should match the snapshot', () => {
    expect( component.html() ).toMatchSnapshot();
  });

  test( 'it unmounts', () => {
    component.unmount();
  } );
} );




