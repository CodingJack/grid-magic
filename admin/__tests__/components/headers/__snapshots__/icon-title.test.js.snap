// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<p
  className="esg-v4-flex-solid esg-v4-pad-top-25"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    style={
      Object {
        "fill": "red",
        "marginLeft": "-4px",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
    />
  </svg>
  <span
    className="esg-v4-pad-left-5"
  >
    Some Text
  </span>
</p>
`;

exports[`it renders correctly 2`] = `
<p
  bold={true}
  className="esg-v4-flex-solid esg-v4-pad-top-25"
  color="green"
  icon="check"
  namespace="name-space"
  text=""
  uppercase={true}
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    style={
      Object {
        "fill": "red",
        "marginLeft": "-4px",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
    />
  </svg>
  <span
    className="esg-v4-pad-left-5"
  >
    Some Text
  </span>
</p>
`;

exports[`it unmounts should match the snapshot 1`] = `"<p class=\\"esg-v4-flex-solid esg-v4-pad-top-25\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" style=\\"fill:red;margin-left:-4px\\"><path d=\\"M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z\\"></path></svg><span class=\\"esg-v4-pad-left-5\\">Some Text</span></p>"`;
